src/gicv3.rs

```diff
@@ -191,13 +198,16 @@ impl IntId {
     }
 
     /// Returns an array of all interrupt Ids that are private to a core, i.e. SGIs and PPIs.
-    pub fn private() -> [IntId; Self::SPI_START as usize] {
-        core::array::from_fn(|i| Self(i as u32))
+    pub fn private() -> impl Iterator<Item = IntId> {
```

Comment:
I still think a `Range<IntId>` would be more useful to return here than just an `impl Iterator`. `Range` has other methods like `contains` that clients might want to use.

Reply:
Ah I see, that's annoying. Never mind then, let's stick with what you have for now. Maybe leave a TODO to return a `Range` once `Step` is stabilised.